src/sidebar/components/annotation-omega.js

```diff
@@ -30,26 +33,27 @@ function AnnotationOmega({
   const draft = useStore(store => store.getDraft(annotation));
   const group = useStore(store => store.getGroup(annotation.group));
 
+  const isPrivate = draft ? draft.isPrivate : !isShared(annotation.permissions);
+  const tags = draft ? draft.tags : annotation.tags;
+  const text = draft ? draft.text : annotation.text;
+
+  const hasQuote = !!quote(annotation);
+  const isEmpty = !text && !tags.length;
+
+  const [isSaving, setIsSaving] = useState(false);
+  const isEditing = !!draft && !isSaving;
+
   useEffect(() => {
     // TEMPORARY. Create a new draft for new (non-highlight) annotations
     // to put the component in "edit mode."
-    if (!draft && isNew(annotation) && !isHighlight(annotation)) {
+    if (!isSaving && !draft && isNew(annotation) && !isHighlight(annotation)) {
```

Comment:
This conditional is getting gross, completely acknowledged. I'm hoping that the whole shebang is temporary, and that the responsibility of creating a draft for an annotation would happen outside the component (seems kind of "duh" when you think about it), i.e. via a service subscribed to the store.

The `!isSaving` addition here is necessary to prevent a race condition in which a new draft gets created for an annotation before the save (API) operation is complete and it is thus assigned an ID (symptom: new annotations re-enter edit mode after they are saved for the first time).